elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/metadata/TableContextTest.java

```diff
@@ -0,0 +1,170 @@
+import com.yahoo.elide.datastores.aggregation.queryengines.sql.annotation.FromTable;
+import com.yahoo.elide.datastores.aggregation.queryengines.sql.dialects.SQLDialectFactory;
+import com.yahoo.elide.datastores.aggregation.queryengines.sql.metadata.SQLReferenceTable;
+import com.yahoo.elide.datastores.aggregation.queryengines.sql.metadata.SQLTable;
+import org.junit.jupiter.api.Test;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+import java.math.BigDecimal;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.persistence.Id;
+import javax.sql.DataSource;
+
+public class TableContextTest {
+
+    private SQLReferenceTable lookupTable;
+    private TableContext revenueFactContext;
+
+    public TableContextTest() {
+        Set<Type<?>> models = new HashSet<>();
+        models.add(ClassType.of(RevenueFact.class));
+        models.add(ClassType.of(CurrencyRate.class));
+
+        EntityDictionary dictionary = new EntityDictionary(new HashMap<>());
+
+        models.stream().forEach(dictionary::bindEntity);
+
+        MetaDataStore metaDataStore = new MetaDataStore(models, true);
+        metaDataStore.populateEntityDictionary(dictionary);
+
+        DataSource mockDataSource = mock(DataSource.class);
+        // The query engine populates the metadata store with actual tables.
+        new SQLQueryEngine(metaDataStore, new ConnectionDetails(mockDataSource, SQLDialectFactory.getDefaultDialect()));
+
+        lookupTable = new SQLReferenceTable(metaDataStore);
+
+        SQLTable revenueFact = metaDataStore.getTable(ClassType.of(RevenueFact.class));
+
+        revenueFactContext = lookupTable.getGlobalTableContext(revenueFact);
+    }
+
+    @Test
+    public void testPhysicalReference() {
+
+        assertEquals("MAX(`com_yahoo_elide_datastores_aggregation_metadata_RevenueFact`.`impressions`)",
+                        revenueFactContext.get("impressions"));
+        assertEquals("`com_yahoo_elide_datastores_aggregation_metadata_RevenueFact_rate`.`provider`",
+                        revenueFactContext.get("rateProvider"));
+    }
+
+    @Test
+    public void testLogicalReference() {
+
+        assertEquals("MAX(`com_yahoo_elide_datastores_aggregation_metadata_RevenueFact`.`impressions`)) * 0.1",
+                        revenueFactContext.get("testImpressions"));
+        assertEquals("TO_CHAR(`com_yahoo_elide_datastores_aggregation_metadata_RevenueFact_rate`.`conversion_rate`, 999D00)",
+                        revenueFactContext.get("conversionRate"));
+    }
+
+    @Test
+    public void testSqlHelper() {
+
+        // default value of 'format' argument is used for 'conversion_rate' column
+        assertEquals("TO_CHAR(SUM(`com_yahoo_elide_datastores_aggregation_metadata_RevenueFact`.`revenue`) * "
+                        + "TO_CHAR(`com_yahoo_elide_datastores_aggregation_metadata_RevenueFact_rate`.`conversion_rate`, 999D00), "
```

Comment:
I don't understand how the conversion rate format here is different than the test below.

Reply:
definition of revenue column uses `{{rate.conversionRate}}` and revenueUsingSqlHelper uses `{{sql from='rate' column='conversionRate'}}`
For first scenario, its using default value of 'format' argument defined for `conversionRate` field in `rate` table.
In the case of SQL helper, its using default value of 'format' argument defined for `revenueUsingSqlHelper` field